### DISCUSSION TITLE
Unable to detect arbitrary colour swatch?

### DISCUSSION THREAD

**Discussion by eKuG** [colour-science/colour-checker-detection#13] [Q&A] [answered]
No colour is being detected. I tried to print the list (Swatches) and the list is empty
![2020-09-13](https://user-images.githubusercontent.com/33174166/93013281-9925ae00-f5c4-11ea-9c8a-97e9c7b119df.png)

**Comment by KelSolaar** [comment]
Hi @eKuG,

Did you try to print or display the masks? Do you have an image of the color checker too?

Cheers,

Thomas

**Comment by eKuG** [comment]
Hi @KelSolaar , I found that no image other than the [Test image here](https://github.com/colour-science/colour-checker-detection/blob/master/colour_checker_detection/examples/examples_detection.ipynb) works.

I tried to print the Swatches, the list was empty. I tried to find other images, The result was the same. 

This is the image I'm working with. Hope you can help me with that!
![FF0000 png](https://user-images.githubusercontent.com/33174166/93014144-215b8180-f5cc-11ea-8c86-def37b98c76a.png)

**Comment by KelSolaar** [comment]
Hello,

So yes it does make sense, this package is about detecting colour checkers, and specifically the ColorChecker Classic from X-Rite, it will not work with random swatches of colour. You need to give it an image that features a ColorChecker Classic, e.g.

![image](https://user-images.githubusercontent.com/99779/93014249-a9a94900-f603-11ea-8127-d3ad9f91ff6c.png)

Cheers,

Thomas

**Comment by eKuG** [comment]
So what I got out of the information that you provided me with is, that with random colours like the one I attached above will not be detected? 
Is there a way to change that somehow? I want to get the color list from a horizontal bar graph. So is it possible to retrain it on a different data set or something? 
Do you have any other way in mind to achieve the above-mentioned goal?

**Comment by KelSolaar** [comment, marked as answer]
> So what I got out of the information that you provided me with is, that with random colours like the one I attached above will not be detected?

Correct! This is not what this repository is intended to do and there is nothing about training here because it is strictly segmentation based.

> Is there a way to change that somehow?

Given what you are trying to achieve, i.e. sampling a regular swatch, you really just need to sample a few pixels in the middle and you are done.

> Do you have any other way in mind to achieve the above-mentioned goal?

Look at the [`swatch_masks`](https://github.com/colour-science/colour-checker-detection/blob/develop/colour_checker_detection/detection/segmentation.py#L133) definition and [how it is used](https://github.com/colour-science/colour-checker-detection/blob/develop/colour_checker_detection/detection/segmentation.py#L773) but keep in mind that what you are trying to do here is totally out of the scope of that repository.

Cheers,

Thomas

**Comment by eKuG** [comment]
Thank you so much for the help @KelSolaar

**Comment by KelSolaar** [comment]
You are welcome!
